notebooks/video_clips/VideoClips.ipynb

```diff
@@ -0,0 +1,963 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "de75fa25-f114-41b3-9b61-d525922f4937",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "video_uuid = str(uuid.uuid5( uuid.NAMESPACE_URL, video_url.format( VIDEO_FILE )))\n",
+    "query=[{\n",
+    "    \"FindVideo\":{\n",
+    "        \"_ref\":1,\n",
+    "        \"constraints\":{\n",
+    "            \"id\": [ \"==\",video_uuid ]\n",
+    "        }\n",
+    "    }\n",
+    "},{\n",
+    "    \"FindFrame\": {\n",
+    "        \"_ref\":2,\n",
+    "        \"uniqueids\":True,\n",
+    "        \"is_connected_to\": {\n",
+    "            \"ref\":1\n",
+    "        },\n",
+    "        \"constraints\": {\n",
+    "            \"_frame_number\": [ \">\", 119 , \"<\",126 ]\n",
+    "        },\n",
+    "        \"results\": {\n",
+    "            \"list\":[\"_frame_number\"]\n",
+    "        }\n",
+    "    }\n",
+    "},{\n",
+    "    \"FindBoundingBox\": {\n",
+    "        \"is_connected_to\": {\n",
+    "            \"ref\":2\n",
+    "        },\n",
+    "        \"group_by_source\":True,\n",
+    "        \"constraints\": {\n",
+    "            \"_label\": [\"==\",\"dog\"]\n",
+    "        },\n",
+    "        \"results\": {\n",
+    "            \"list\":[\"confidence\"]\n",
+    "        }\n",
+    "    }\n",
+    "}]\n",
+    "\n",
+    "r,_=c.query(query)\n",
+    "print(r)\n",
+    "            "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "33deeabb-bd05-48c0-91ee-e7a718f10c75",
+   "metadata": {},
+   "source": [
+    "That output is a bit hard to read, so let's pull out the information from the json in an easy to understand format."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8676500d-cded-4d9f-8a26-76c9b9ab9285",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# reference the results from the bounding box.\n",
+    "dog_results=r[2][\"FindBoundingBox\"][\"entities\"]\n",
+    "# reference the results from the frames.\n",
+    "frame_results=r[1][\"FindFrame\"][\"entities\"]\n",
+    "\n",
+    "# bounding boxes are referenced to their frame objects by the database unique id, so we'll create a map\n",
+    "# so we can get the frame data for each detection.\n",
+    "frame_map = { fr['_uniqueid']: fr['_frame_number'] for fr in frame_results }\n",
+    "\n",
+    "for dog_detection in  dog_results.keys():\n",
+    "    print(f\"In Frame {frame_map[dog_detection]} dog confidence was {dog_results[dog_detection][0]['confidence']}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "710f58f2-bbc5-4fc2-bcab-dba1bbca01c8",
+   "metadata": {},
+   "source": [
+    "## Verification Solution\n",
+    "Now we've been able to tell with a powerful query that we didn't have a dog confidence high enough to start a clip.\n",
+    "\n",
+    "We can then either decide to fine tune or model - have it train on partially obscured dogs more, lower the threshold, or accept our results.\n",
```

Comment:
Where should people go next from here? Put some suggested links (remember docs :) )